zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/deployment/ProcessStateTest.java

```diff
@@ -1041,6 +1048,45 @@ public void shouldAddProcessAfterOnlyVersionIsDeleted() {
     assertThat(processState.getNextProcessVersion(processId, TENANT_ID)).isEqualTo(3);
   }
 
+  @Test
+  public void shouldIterateThroughAllProcesses() {
+    // given
+    final var process1 = creatingProcessRecord(processingState, "process1", 1);
+    final var process2 = creatingProcessRecord(processingState, "process2", 1);
+    final var process3 = creatingProcessRecord(processingState, "process3", 1);
+    processState.putProcess(process1.getKey(), process1);
+    processState.putProcess(process2.getKey(), process2);
+    processState.putProcess(process3.getKey(), process3);
+    final var visitor = mock(PersistedProcessVisitor.class);
+    when(visitor.visit(any())).thenReturn(true);
+
+    // when
+    processState.forEachProcess(null, visitor);
+
+    // then
+    verify(visitor, times(3)).visit(any());
```

Comment:
It's a bit ugly because we can't just verify based on the argument because iteration does not make a copy. But it makes sense to be stricter here :+1: 
I've added an intermediary visitor to copy out the key and verify only that now. 53aeabd

Reply:
Hm, the argument captor would need to either make a copy or only capture the key. What I have written here is basically the same.